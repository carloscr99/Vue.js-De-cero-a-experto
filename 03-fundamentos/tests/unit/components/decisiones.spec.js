import {shallowMount} from '@vue/test-utils'
import Decisiones from '@/components/Decisiones.vue'
import { createLogicalAnd } from 'typescript';

describe('Decisiones Component', ()=>{

    let wrapper;
    let consoleLogSpy;
    
    beforeEach(()=> {
        wrapper = shallowMount(Decisiones)

        consoleLogSpy = jest.spyOn(console, 'log')

    })


    test('debe de hacer match', ()=>{

            expect (wrapper.html()).toMatchSnapshot()
        
    })

    test('Escribir en el input no ejecuta', async()=>{

        const input = wrapper.find('input')
        await input.setValue('Hola mundo')

        expect( consoleLogSpy ).toHaveBeenCalledTimes(2)

    })

    test('Escribir ? dispara el fetch',()=>{

        

    })

    test('pruebas en getAnswer',()=>{

        

    })

    test('pruebas en getAnswer - Fallo en el API',()=>{

        

    })

})
