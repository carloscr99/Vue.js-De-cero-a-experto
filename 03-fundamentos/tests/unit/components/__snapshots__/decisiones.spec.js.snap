// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Decisiones Component debe de hacer match 1`] = `
<h1>Indecision</h1>
<!--v-if-->
<div class="bg-dark"></div>
<div class="indecision-container"><input type="text" placeholder="Hazme una pregunta">
  <p>Para finalizar, introduce el signo (?)</p>
  <!--v-if-->
</div>
`;
